### PR TITLE
Run pre-commit in GitHub Actions

### PR DESCRIPTION
**Pull request by KyleFromNVIDIA**

### CODE REVIEW

File: ci/lint.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+# Runs pre-commit
+
+set -ue
+
+pip install pre-commit
+
+pre-commit run -a
```
Comment (bdice) [inline ci/lint.sh:8]:
```suggestion
pre-commit run --all-files --show-diff-on-failure
```

**Review by bdice — APPROVED** [review]
One suggestion, otherwise LGTM.

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+      - name: Lint
```
Comment (raydouglass) [inline .github/workflows/lint.yaml:24]: Could we use https://github.com/pre-commit/action instead? It's already added to the allow-list for 3rd party actions.

Comment (KyleFromNVIDIA) [reply .github/workflows/lint.yaml:24]: From https://github.com/pre-commit/action:

> this action is in maintenance-only mode and will not be accepting new features.
>
> generally you want to use [pre-commit.ci](https://pre-commit.ci/) which is faster and has more features.

Could we use pre-commit.ci instead?

**Review by raydouglass — APPROVED** [review]
Since https://github.com/pre-commit/action is in maintenance mode, I think the current approach is fine.

**Pull request merged**